utils/python/CIME/case_st_archive.py

```diff
@@ -1,24 +1,413 @@
+"""
+short term archiving
+"""
+
 from CIME.XML.standard_module_setup import *
 from CIME.case_submit               import submit
 from CIME.case                      import Case
+from CIME.XML.env_archive           import EnvArchive
 from CIME.utils                     import get_model, run_cmd, append_status
+from CIME.utils                     import expect, append_status
+from os.path                        import isfile, isdir, join
+import shutil, glob, re
 
 logger = logging.getLogger(__name__)
 
 ###############################################################################
+def check_run(case):
+###############################################################################
+
+    logger.debug('In check_run...')
+    dout_s_root = case.get_value('DOUT_S_ROOT')
+    if dout_s_root is None or dout_s_root == 'UNSET':
+        expect(False, 'XML variable DOUT_S_ROOT is required for root location of short-term achiver')
+    if not isdir(dout_s_root):
+        os.makedirs(dout_s_root)
+    dout_s_save_interim = case.get_value('DOUT_S_SAVE_INTERIM_RESTART_FILES')
+    if dout_s_save_interim == 'FALSE' or dout_s_save_interim == 'UNSET':
+        rest_n = int(case.get_value('REST_N'))
```

Comment:
These options are known to be integer types so will be returned as ints. There's no need for the casts